src/main/java/fr/cgi/minibadge/service/impl/DefaultUserService.java

```diff
@@ -61,4 +75,49 @@ private Future<JsonArray> searchRequest(HttpServerRequest request, String query)
 
         return promise.future();
     }
+
+    @Override
+    public Future<Void> upsert(List<String> usersIds) {
+        Promise<Void> promise = Promise.promise();
+        JsonArray statements = new JsonArray();
+        Set<String> distinctUsersIds = new HashSet<>(usersIds);
+        usersIds = new ArrayList<>(distinctUsersIds);
+        JsonObject params = new JsonObject();
+        params.put(USER_ID, new JsonArray(usersIds));
+        params.put(ACTION, LIST_USERS);
+        List<User> users = new ArrayList<>();
+        eb.request(DIRECTORY, params, handlerToAsyncHandler(message -> {
+            if (message.body().getString(STATUS).equals(OK)) {
+                message.body().getJsonArray(RESULT).stream().forEach(userData -> {
+                    User user = new User();
+                    JsonObject userJo = ((JsonObject) userData);
+                    user.setUserId(userJo.getString(Field.ID));
+                    user.setUsername(userJo.getString(Field.USERNAME));
+                    users.add(user);
+                });
+                users.forEach(user -> statements.add(upsertStatement(user)));
+                sql.transaction(statements, event -> {
+                    if (event.body().getString(STATUS).equals(OK)) {
+                        promise.complete();
+                    }
+                });
+            }
+        }));
+        return promise.future();
```

Comment:
Comme je te disais dans mon commentaire plus tôt, je l'ai déjà implémenté:

```suggestion
@Override
    public Future<List<User>> getUsers(List<String> userIds) {
        Promise<List<User>> promise = Promise.promise();

        getUsersRequest(userIds)
                .onFailure(promise::fail)
                .onSuccess(users -> promise.complete(new User().toList(users)));

        return promise.future();
    }

    private Future<JsonArray> getUsersRequest(List<String> userIds) {
        Promise<JsonArray> promise = Promise.promise();

        JsonObject action = new JsonObject()
                .put(EventBusConst.ACTION, EventBusConst.LIST_USERS)
                .put(Field.USERIDS, userIds);

        eb.request(EventBusConst.DIRECTORY, action, PromiseHelper.messageHandler(promise, "")); //todo

        return promise.future();
    }
```

Fait attention, a bien distinguer tes besoins. Dans l'idéal, c'est une fonction, un besoin.